### ISSUE TITLE
`Error: invalid topic name` when creating a FIFO Topic

### ISSUE THREAD

**Issue by sfenman** [terraform-aws-modules/terraform-aws-sns#54]
## Description

When enabling FIFO SNS topic by using:

```
  fifo_topic                  = true
  content_based_deduplication = true
```

If I don't append the `.fifo` suffix in my topic name it throws an error: `Error: invalid topic name: my-topic`.

In [terraform-aws-sqs](https://github.com/terraform-aws-modules/terraform-aws-sqs) module the `.fifo` suffix is appended automatically when creating a FIFO queue.

Let me know if I can open a PR with the change in case you think it would be better to do something similar for this module too.

**Comment by antonbabenko** [comment]
SQS queue must be FIFO as well. Please check an existing example in https://github.com/terraform-aws-modules/terraform-aws-sns/blob/master/examples/complete/main.tf

**Comment by sfenman** [comment]
Right I am using an SQS FIFO as well. What I meant is something similar to what the [terraform-aws-sqs](https://github.com/terraform-aws-modules/terraform-aws-sqs/blob/8c18f70fd765db2adf31edf5fc15b3058367e5a2/main.tf#L25) module does.

In the example you mentioned and if I set the `use_name_prefix` to false I get the same error:

<img width="555" alt="image" src="https://github.com/terraform-aws-modules/terraform-aws-sns/assets/43748066/938a1dbc-2390-4777-9271-205290e78f9d">

**Comment by antonbabenko** [comment]
This issue has been resolved in version 6.1.0 :tada:
